src/states/node.rs

```diff
@@ -433,14 +433,17 @@ impl Node {
             .iter()
             .filter(|event| match **event {
                 // Only re-vote not yet accumulated events and still relevant to our new prefix.
-                NetworkEvent::Online(pub_id, _) | NetworkEvent::Offline(pub_id) => {
+                NetworkEvent::Offline(pub_id) => {
                     our_pfx.matches(pub_id.name()) && !completed_events.contains(event)
                 }
 
-                // Only re-vote not yet accumulated events and still relevant to our new prefix.
-                NetworkEvent::ExpectCandidate(ExpectCandidatePayload { ref dst_name, .. }) => {
-                    our_pfx.matches(dst_name) && !completed_events.contains(event)
-                }
+                // Drop candidates that have not completed:
+                // Called peer_manager.remove_candidate reset the candidate so it can be shared by
+                // all nodes: Because new node may not have voted for it, Forget the votes in
+                // flight as well.
+                NetworkEvent::Online(_, _)
+                | NetworkEvent::ExpectCandidate(_)
+                | NetworkEvent::PurgeCandidate(_) => false,
```

Comment:
I'm confused about this. Why is it OK to ignore these events now?

Reply:
more so just not carrying candidate info into a new prefix than ignoring. currently candidate info isnt part of the shared state, so you could have a pfx change with members having different candidates, to prevent that, we're resetting candidate info on pfx change and thus these in flight events that also influence the old candidate state from the old prefix are filtered out to not be replayed in the new prefix to prevent it from affecting a new candidate in the next prefix. thus giving all nodes entering that prefix the same blank state if that helps.

Reply:
OK. But isn't the old candidate going to be forgotten now? Shouldn't we at least inform them somehow that they should repeat the join attempt with the new prefix?

Reply:
They can't, we are not waiting for them anymore. At this point they are gone from the network and have to rejoin when they time out on their side. (This is not something that should happen in the final design).

Reply:
If we send them a message to restart the candidate process it would speed things up as they wouldn't have to wait for the timeout. Probably not important enough to deserve complicating the code though. I'm OK with it.

Reply:
yeh the timeout from joining node side being the key ^^ cos even without a pfx change, we dont inform a joining node they failed res-proof/join process. thats just left to their timeout of not getting a NodeApproval ultimately to switch to establishing state and that handles these too for now while eventually as jp mentioned the candidate info itself with node-ageing wouldnt be ignored like now.